internal/security/secretstoreclient/vault.go

```diff
@@ -272,3 +161,49 @@ func (vc *vaultClient) commonRequest(token *string, method string, url string, j
 	req.Header.Set("Content-Type", JSONContentType)
 	return vc.client.Do(req)
 }
+
+// processResponse takes the output of the commonRequest method
+// and performs standard processing based the the following:
+// operationDescription contains a message describing the operation in logs
+// expectedStatusCode is the expected status code for success
+// responseObject is the address of a JSON struct, or nil
+// If the response is HTTP 200 status and responseObject is not nil,
+// this function will attempt to decode the object
+// delegate is a callback function called to handle the decoded response
+func (vc *vaultClient) processResponse(resp *http.Response, responseError error,
+	operationDescription string, expectedStatusCode int, responseObject interface{},
+	delegate func() error) (statusCode int, err error) {
+
+	if responseError != nil {
+		vc.logger.Error(fmt.Sprintf("unable to make request to %s failed: %s", operationDescription, err.Error()))
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != expectedStatusCode {
+		vc.logger.Error(fmt.Sprintf("request to %s failed with status: %s", operationDescription, resp.Status))
+		return resp.StatusCode, err
+	}
+
+	if responseObject != nil && resp.StatusCode == http.StatusOK {
```

Comment:
how does this work if I pass `expectedStatusCode` as http.StatusAccepted? Then this won't decode anything into the responseObject AFAICT?

Reply:
Which API call returns http.StatusAccepted with JSON body on success?

Reply:
Addressed in #1855.  Found one API after grep of Vault source where StatusAccepted was returned with a JSON response.